### PR TITLE
compose: Omit |<id> in quote-and-reply mention unless mention is ambiguous

### PR DESCRIPTION
**Pull request by chrisbobbe**

### CODE REVIEW

File: lib/model/compose.dart [hunk 1]
```diff
@@ -226,6 +235,9 @@ String quoteAndReplyPlaceholder(PerAccountStore store, {
 ///     ```quote
 ///     message content
 ///     ```
+///
+/// When calling [mention] for the silent @-mention of the author, this
+/// passes [store.users], opting into a linear scan through all users.
 String quoteAndReply(PerAccountStore store, {
```
Comment (gnprice) [inline lib/model/compose.dart:241]: Hmm, good point about the linear scan.

The quote-and-reply use case actually strikes me as one where the extra noise of the user ID doesn't matter — there's already that generally-super-long link there, so this is generally a line that users are going to leave alone and not be looking closely at the insides of. Perhaps it's therefore best to leave the user ID in unconditionally in this case, and only seek to leave it out when autocompleting an @-mention.

(Which means, for this PR, adding the feature to `mention` and to that function's unit tests.)

In fact, at a quick experiment, that seems to be web's behavior! It seems to always include the user ID in a quote-and-reply attribution line, even for users where it doesn't in an autocompleted @-mention.

**Review by gnprice — COMMENTED** [review]
Thanks! Glad to have this so quickly.

File: lib/model/compose.dart (hunk 1 above)
Comment (chrisbobbe) [reply lib/model/compose.dart]: Makes sense.

**Comment by chrisbobbe** [comment]
Thanks for the review! Revision pushed.

**Comment by gnprice** [comment]
Thanks! Looks good; merging.

**Pull request merged**